src/engine/shape-builder.js

```diff
@@ -338,11 +394,34 @@ export default class ShapeBuilder extends DrawEngine {
       if (selectedShape && !selectedShape.selected) {
         const selectEventObj = selectShape(selectedShape, shapes, this._selectedShapes, this._selectStyle, this._xformStyle, selectedInfo)
         this.fire(EventConstants.SELECTION_CHANGED, selectEventObj)
+      } else {
+        // If user clicks anywhere outside then allow the movement of Base Map (Parents Container)
+        this._makeParentElementMovable()
+      }
+    }
+    // Added Support for Double click
+    if (event.touches) {
+      if (event.cancelable) {
+        event.preventDefault()
+      }
+      if ((Date.now() - this.firstTapTime) < DOUBLE_CLICK_DELAY) {
+        this._dblclickCB(event)
       }
     }
+    this.firstTapTime = Date.now()
   }
 
   _mousemoveCB(event) {
+    this.setDenyMouseEventFlag(event)
+    if (this.denyMouseEvent && !event.touches) {
```

Comment:
there's a lot of checks for `event.touches` since you've collapsed the mouse/touch callbacks together. I think it'd be a bit more readable to turn this into a utility function: `isTouchEvent(event)`. It would return true if the event was a touch event.